krm-functions/lib/condkptsdk/sdk.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package condkptsdk
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	kptfilelibv1 "github.com/nephio-project/nephio/krm-functions/lib/kptfile/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const FnRuntimeOwner = "fnruntime.nephio.org/owner"
+const FnRuntimeDelete = "fnruntime.nephio.org/delete"
+
+type KptCondSDK interface {
+	Run() (bool, error)
+}
+type ResourceKind string
+
+const (
+	// ChildRemoteCondition defines a GVK resource for which only conditions need to be created
+	ChildRemoteCondition ResourceKind = "remoteCondition"
+	// ChildRemote defines a GVK resource for which conditions and resources need to be created
+	ChildRemote ResourceKind = "remote"
+	// ChildLocal defines a GVK resource for which no conditions need to be created
+	ChildLocal ResourceKind = "local"
+)
+
+type Config struct {
```

Comment:
I have tried to do this before but since you execute in the pkgname of the sdk I feel that it is ok to use Config since you configure the SDK. Thoughts?

Reply:
sure, that's fine